packages/eslint-plugin-next/src/rules/no-before-interactive-script-outside-document.ts

```diff
@@ -25,24 +22,17 @@ export = defineRule({
 
     return {
       'ImportDeclaration[source.value="next/script"] > ImportDefaultSpecifier'(
-        node
+        node: any
       ) {
         scriptImportName = node.local.name
       },
       JSXOpeningElement(node) {
-        let pathname = context.getFilename()
+        const pathname = convertToCorrectSeparator(context.getFilename())
 
-        if (startsWithUsingCorrectSeparators(pathname, 'src/')) {
-          pathname = pathname.slice(4)
-        } else if (startsWithUsingCorrectSeparators(pathname, '/src/')) {
-          pathname = pathname.slice(5)
-        }
+        const isInAppDir = pathname.includes(`app${path.sep}`)
```

Comment:
Seems it should check this way?
```suggestion
        const isInAppDir = pathname.includes(`${path.sep}app${path.sep}`)
```

Reply:
Fixed & tested it, thank you!